rustls/src/crypto/mod.rs

```diff
@@ -45,19 +45,137 @@ pub use crate::rand::GetRandomFailed;
 
 pub use crate::msgs::handshake::KeyExchangeAlgorithm;
 
-/// Pluggable crypto galore.
+/// Controls core cryptography used by rustls.
+///
+/// This crate comes with two built-in options, provided as
+/// `&dyn CryptoProvider` values:
+///
+/// - [`crate::crypto::ring::RING`]: (behind the `ring` crate feature, which
+///   is enabled by default).  This provider uses the [*ring*](https://github.com/briansmith/ring)
+///   crate.
+/// - [`crate::crypto::aws_lc_rs::AWS_LC_RS`]: (behind the `aws_lc_rs` feature,
+///   which is optional).  This provider uses the [aws-lc-rs](https://github.com/aws/aws-lc-rs)
+///   crate.
+///
+/// # Using a specific `CryptoProvider`
```

Comment:
Style nit: I feel like these headers stand out more (and thus help reading/skimming) if there's an empty line before and after, instead of only before.